runtime/cudaq/algorithms/observe.h

```diff
@@ -493,21 +493,13 @@ auto observe_async(const std::size_t qpu_id, QuantumKernel &&kernel, spin_op &H,
   auto shots = platform.get_shots().value_or(-1);
   auto kernelName = cudaq::getKernelName(kernel);
 
-#if CUDAQ_USE_STD20
   return details::runObservationAsync(
-      [&kernel, ... args = std::forward<Args>(args)]() mutable {
-        cudaq::invokeKernel(std::forward<QuantumKernel>(kernel),
-                            std::forward<Args>(args)...);
-      },
+      detail::make_copyable_function(
+          [&kernel, ... args = std::forward<Args>(args)]() mutable {
```

Comment:
How does this syntax compile as C++17? It's using the same perfect forwarding syntax as the C++20 code without the ifdef guard.

Reply:
Are we sure this template is being instantiated in the C++17 tests?

Reply:
I'm pretty sure that these functions are used in the `targettests/Remote-Sim/observe_async.cpp` and `targettests/Remote-Sim/sample_async.cpp` tests, which are both tested with `-std=c++17` for the installer CI.

I think the `make_copyable_function` helper does the trick in a way that is compatible with C++17.

Reply:
The change above has been reverted, i.e., keeping two different paths for C++17 and C++20. The pre-existing compiler error in the C++17 path is fixed by using the `make_copyable_function` helper.

Also, the unit tests (`Remote-Sim/observe_async.cpp` and  `Remote-Sim/sample_async.cpp`) are updated to exercise all code paths (the two overloads of `observe_async` and `sample_async`)